flask_multipass_saml_groups/group_provider/sql.py

```diff
@@ -0,0 +1,123 @@
+#  Copyright 2023 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""A group provider that persists groups and their members in a SQL database provided by Indico."""
+
+from typing import Iterable, Iterator, Optional
+
+from flask_multipass import Group, IdentityInfo, IdentityProvider
+from indico.core.db import db
+
+from flask_multipass_saml_groups.group_provider.base import GroupProvider
+from flask_multipass_saml_groups.models.saml_groups import SAMLGroup as DBGroup
+from flask_multipass_saml_groups.models.saml_groups import SAMLUser
+
+
+class SQLGroup(Group):
+    """A group whose group membership is persisted in a SQL database.
+
+    Attrs:
+        supports_member_list (bool): If the group supports getting the list of members
+    """
+
+    supports_member_list = True
+
+    def __init__(self, provider: IdentityProvider, name: str):
+        """Initialize the group.
+
+        Args:
+            provider: The associated identity provider
+            name: The name of this group
+        """
+        super().__init__(provider, name)
+        self._provider = provider
+        self._name = name
+
+    def get_members(self) -> Iterator[IdentityInfo]:  # noqa: D102,DCO010 docstring in base class
+        db_group = DBGroup.query.filter_by(name=self._name).first()
```

Comment:
Do you think this query might need to be case-insensitive?

Reply:
No, group names are case-sensitive and this method will only return the members of that particular group.

Reply:
Might be worth a docstring for this method, and you could explain there that group names are case-sensitive.